### PR TITLE
I added the address entity!

### PR DESCRIPTION
**Pull request by aureliosss**
I added a sequence for the table!
I added the OneToOne relationship between app_user and address! 
I added the query in script.sql!
image

HTML updates must be done!

### CODE REVIEW

**Review by gonciarstef — COMMENTED** [review]
Looks good!

File: src/main/resources/db/scripts.sql [hunk 1]
```diff
@@ -42,14 +42,31 @@ CREATE TABLE IF NOT EXISTS app_user
...
-VALUES (3, 'Jane',  'Dow','JaneUser', 'pass123', 'jane@email.com');
+VALUES (3, 'Jane', 'Dow', 'JaneUser', 'pass123', 'jane@email.com');
 INSERT INTO app_user(first_name, last_name, username, password, email)
 VALUES ('Elena', 'Tintiri', 'ElenaUser', 'pass123', 'elena@email.com');
 INSERT INTO app_user(fk_role, first_name, last_name, username, password, email)
 VALUES (2, 'David', 'Beckham', 'DavUser', 'pass123', 'david@email.com');
+
+CREATE TABLE IF NOT EXISTS address
+(
+    id                 BIGINT      NOT NULL DEFAULT NEXTVAL('seq_address'),
```
Comment (Dariusionut) [inline src/main/resources/db/scripts.sql:59]: Where is seq_address defined? It does not exist and the scripts cannot be run

File: src/main/resources/db/scripts.sql [hunk 2]
```diff
@@ -42,14 +42,31 @@ CREATE TABLE IF NOT EXISTS app_user
...
+    country            VARCHAR(45) NOT NULL,
+    city               VARCHAR(45) NOT NULL,
+    door_number        VARCHAR(10) NOT NULL,
+    additional_details VARCHAR(65) NULL,
+    PRIMARY KEY (id)
```
Comment (Dariusionut) [inline src/main/resources/db/scripts.sql:64]: CONSTRAITN address_pk PRIMARY KEY (pk). Please follow the same design - look at the app_user table

File: src/main/resources/db/scripts.sql [hunk 3]
```diff
@@ -42,14 +42,31 @@ CREATE TABLE IF NOT EXISTS app_user
...
+);
+CREATE SEQUENCE IF NOT EXISTS seq_address;
+
+ALTER TABLE app_user
+    ADD COLUMN fk_address BIGINT UNIQUE,
+    ADD CONSTRAINT fk_address
```
Comment (Dariusionut) [inline src/main/resources/db/scripts.sql:70]: Constraint name should be (table name + column name where you put the constraint) -> app_user_fk_address

File: src/main/java/com/spring/studentmanagement/models/Address.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    private String country;
+
+    @Column(name = "city", length = 45, nullable = false)
+    private String city;
+
+    @Column(name = "door_number", length = 10, nullable = false)
+    private String doorNumber;
+
+    @Column(name = "additional_details")
+    private String additionalDetails;
```
Comment (Dariusionut) [inline src/main/java/com/spring/studentmanagement/models/Address.java:34]: Also specify the length. The JPA entities must reflect the SQL tables

File: src/main/java/com/spring/studentmanagement/models/Address.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import jakarta.persistence.*;
+import lombok.*;
+
+/**
+ * Created at 23.04.2023 by Dan.
+ */
+@Getter
+@Setter
+@ToString
+@RequiredArgsConstructor
```
Comment (Dariusionut) [inline src/main/java/com/spring/studentmanagement/models/Address.java:12]: Do we need @RequiredArgConstructor? YAGNI principle - You Are Not Gonna Need It. Do not write code unless you are using it

File: src/main/java/com/spring/studentmanagement/models/Role.java [hunk 6]
```diff
@@ -23,4 +25,14 @@ public class Role {
 
     @Column(name = "role_name", length = 45, unique = true, nullable = false)
     private String roleName;
+    @OneToMany(mappedBy = "role")
+    private Collection<AppUser> appUser;
+
+    public Collection<AppUser> getAppUser() {
+        return appUser;
```
Comment (Dariusionut) [inline src/main/java/com/spring/studentmanagement/models/Role.java:32]: To be removed
Why this? This is a Bi directional relation which we don't need.
The getAppUser and setAppUser methods are not used in the program.
YAGNI principle.

File: src/main/java/com/spring/studentmanagement/models/Role.java [hunk 7]
```diff
@@ -23,4 +25,14 @@ public class Role {
...
+    }
+
+    public void setAppUser(Collection<AppUser> appUser) {
```
Comment (Dariusionut) [inline src/main/java/com/spring/studentmanagement/models/Role.java:35]: To be removed. YAGNI

File: src/main/java/com/spring/studentmanagement/models/Role.java [hunk 8]
```diff
@@ -23,4 +25,14 @@ public class Role {
 
     @Column(name = "role_name", length = 45, unique = true, nullable = false)
     private String roleName;
+    @OneToMany(mappedBy = "role")
+    private Collection<AppUser> appUser;
```
Comment (Dariusionut) [inline src/main/java/com/spring/studentmanagement/models/Role.java:29]: We don't need bi-directional relation. We only access the role through the AppUser entity

**Review by Dariusionut — CHANGES_REQUESTED** [review]

**Review by Dariusionut — APPROVED** [review]

**Comment by Dariusionut** [comment]
"I added the address entity!" is not a valid pull request title!
Please be professional
**Use ticket name instead**

**Pull request merged**